src/components/UsersDetailsCard.jsx

```diff
@@ -33,10 +33,45 @@ const tableHeaders = [
   },
 ];
 
+function reducer(state, action) {
+  return { ...state, ...action };
+}
+
 export default function UsersDetailsCard() {
   const { enqueueSnackbar } = useSnackbar();
 
-  const [selectedArray, setSelectedArray] = useState([]);
+  const [
+    { selectedArray, searchText, open, selectAll, buttonArray, selectedButton },
```

Comment:
give meaning full names, like ```selectedArray``` if it holds  user id in it then its better to name it as ```selectedUserIds```.
same goes with buttonArray, selectedButton.
try to give meaning full names to variables.

Reply:
done